include/pulsar/ConsumerConfiguration.h

```diff
@@ -383,6 +383,26 @@ class PULSAR_PUBLIC ConsumerConfiguration {
      */
     InitialPosition getSubscriptionInitialPosition() const;
 
+    /**
+     * Selects the subscription mode to be used when subscribing to a topic.
+     *
+     * <p>Options are:
+     * <ul>
+     *  <li>{@link SubscriptionMode#Durable} (Default)</li>
+     *  <li>{@link SubscriptionMode#NonDurable}</li>
+     * </ul>
+     *
+     * @param subscriptionMode the subscription mode value
+     */
+    ConsumerConfiguration& setSubscriptionMode(SubscriptionMode subscriptionMode);
+
+    /**
+     * Get subscription mode.
+     *
+     * @return
+     */
+    SubscriptionMode getSubscriptionMode() const;
```

Comment:
It's better not support setting the subscription mode. /cc @codelipenghui

Reply:
done.